src/app/contacts/contact/contact.component.ts

```diff
@@ -0,0 +1,16 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ContactType } from '../contact-list/contact-list.component';
+
+@Component({
+  selector: 'app-contact',
+  templateUrl: './contact.component.html',
+  styleUrls: ['./contact.component.css']
+})
+export class ContactComponent implements OnInit{
+  contact: ContactType | null = null
+
+  ngOnInit(): void {
+    this.contact = history.state
```

Comment:
We could also get the contact from the `ContactsService`. There might be a problem with `history.state` if a user navigates to the contact from a "fresh" browser